pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Validate Messages/Events in runtime
+description: This guide explains different use cases for messages/events validation using AsyncAPI.
+weight: 120
+---
+
+# Introduction
+In this guide, you will learn how to validate the messages in in runtime after creating your application using AsyncAPI.
+
+# Message validation
+In order to understand message validation, we need to first understand the basic components involved.
+- Producer: The producer is responsible for producing the messages.
+- Consumer: The consumer is responsible for getting the producer's messages.
+- Broker: The broker acts as a bridge in between the consumer and the producer. The messages travel through the broker.
+Validation can be done at producer and consumer. Consumer side validation can happen by discarding invalid messages based on parameters.
+The consumer and producer cannot communicate directly and here AsyncAPI file tells what should be included in the payload when a service produces a message and also tells the consumer about the properties in the message.
```

Comment:
```suggestion
Because consumers and producers cannot communicate directly, the AsyncAPI file dictates what should be included in the payload when a service produces a message. The AsyncAPI file also tells the consumer about the message's properties.
```